qos/solana/endpoint.go

```diff
@@ -53,3 +55,22 @@ func (e endpoint) ValidateBasic() error {
 		return nil
 	}
 }
+
+// ApplyObservation updates the data stored regarding the endpoint using the supplied observation.
+// It Returns true if the observation was non-generic, i.e. mutated the endpoint.
+// TODO_TECHDEBT: add a method to distinguish a bad endpoint, i.e. an endpoint which failed to respond to a request, from an endpoint with no/incomplete observations.
```

Comment:
Please add your name (or other) to TECHDEBT we actually plan on tackling so we have a DRI.